src/yoroi-wallets/cardano/addressInfo.ts

```diff
@@ -0,0 +1,87 @@
+import {WasmContract} from '@emurgo/yoroi-lib-core'
+
+import {Address, BaseAddress, RewardAddress} from '.'
+
+export type CardanoKeyHashes = {
+  spending: string | null
+  staking: string | null
+}
+
+/**
+ * @description Get the spending keyHash, resolves null for PointerAddress & EnterpriseAddress missing yoroi-lib impl
+ *
+ * @param {WasmContract.Address} address wasm address
+ * @returns Promise<WasmContract.Ed25519KeyHash | undefined> null for legacy/other addresses and undefined for scriptHash
+ */
+async function getSpendingKeyHash(
+  address: WasmContract.Address,
+): Promise<WasmContract.Ed25519KeyHash | null | undefined> {
+  const baseAddr = await BaseAddress.fromAddress(address)
+  if (baseAddr.hasValue()) return baseAddr.paymentCred().then((paymentCred) => paymentCred.toKeyhash())
+
+  const rewardAddr = await RewardAddress.fromAddress(address)
+  if (rewardAddr.hasValue()) return rewardAddr.paymentCred().then((paymentCred) => paymentCred.toKeyhash())
+
+  return null
+}
+
+/**
+ * @description Get the staking keyHash of a BaseAddress
+ *
+ * @param {WasmContract.Address} address
+ * @returns {Promise<WasmContract.Ed25519KeyHash | null | undefined>} null for legacy/other addresses and undefined for scriptHash
+ */
+async function getStakingKeyHash(
+  address: WasmContract.Address,
+): Promise<WasmContract.Ed25519KeyHash | null | undefined> {
+  const baseAddr = await BaseAddress.fromAddress(address)
+  if (baseAddr.hasValue()) return baseAddr.stakeCred().then((paymentCred) => paymentCred.toKeyhash())
+
+  return null
+}
+
+async function toHexKeyHash(keyHash: WasmContract.Ed25519KeyHash | null | undefined): Promise<string> {
+  if (!keyHash) return ''
+  if (!keyHash.hasValue()) return ''
+
+  return keyHash.toBytes().then((bytes) => Buffer.from(bytes ?? '').toString('hex'))
+}
+
+/**
+ * @description Try to resolve the spending and staking key hashes for a bech32 address
+ *
+ * @param {string} address expects to be a bech32
+ * @returns {Promise<CardanoKeyHashes>}
+ */
+export async function getKeyHashes(address: string): Promise<CardanoKeyHashes> {
```

Comment:
should these be split up and loaded separately?

Reply:
it does

Reply:
i meant, 2 different hooks, then you can move them into separate components with separate boundaries

Reply:
I think we will always display them together, we are using it only in 1 place, but now I get your point, will do one more iteration, then when lib comes out with the hooks will be ready.